File: ex_2/SET-5/app/src/main/java/com/example/socialmediapost/MainActivity.java
package com.example.socialmediapost;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
public class MainActivity extends AppCompatActivity {
    private int count = 0;
    private boolean isLiked = false;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button likeButton = findViewById(R.id.likeButton);
        final TextView likeCounter = findViewById(R.id.likeCounter);
        likeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!isLiked) {
                    count++;
                    isLiked = true;
                    likeButton.setText("Liked");
                    likeCounter.setText("Likes: " + count);
                }
            }
        });
    }
}

File: ex_1/SET-1/ChangeofclrofTextview/app/src/main/java/com/example/changeofclroftextview/MainActivity.kt
package com.example.changeofclroftextview
import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlin.properties.Delegates

class MainActivity : AppCompatActivity() {
    private var contentView by Delegates.notNull<Int>()
    private var colorIndex = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val colorTextView = findViewById<TextView?>(R.id.colorTextView)
        val changeColorButton = findViewById<Button?>(R.id.changeColorButton)
        val colors = intArrayOf(
            Color.RED,
            Color.GREEN,
            Color.BLUE,
            Color.YELLOW,
            Color.DKGRAY,
            Color.CYAN,
            Color.MAGENTA
        )
        changeColorButton?.setOnClickListener {
            colorTextView?.setBackgroundColor(colors[colorIndex])
            colorIndex = (colorIndex + 1) % colors.size
        }
    }
}